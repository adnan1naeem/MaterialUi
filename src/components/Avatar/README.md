# Avatar

Avatars are found throughout material design with uses in everything from tables to dialog menus.

## Compatibility

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :----: | :---------: | :-------------: |
| ✔️      | ✖            | ✖              |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :---: | :--------: | :------------: |
| ![web image](./screenshots/Avatar.png) |    TBD   | TBD |

## Universal Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| style | Object |  | Get the style object |
| src | string | | src of image. |
| children | Node | | Children of avatar. |

## Specific Library Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| alt | string |  | string to be shown if image was not rendered due to any error. |
| classes | object | | classes prop to override styling. |
| component | string or function | 'div' | Component with which the avatar will be rendered |
| imgProps | Object |  | Props of image tag. |
| sizes | string | | sizes of image. |

## How to use

```JavaScript
import React from 'react';
import BR from '@blueeast/bluerain-os';

const AvatarSample = (props) => {
    const Avatar = BR.Components.get('Avatar');
    return (
        <Avatar
        alt="Remy Sharp"
        src="https://material-ui-next.com/static/images/remy.jpg"
        style={{height:35,width:35}}
        />);
};

export default AvatarSample;
```
