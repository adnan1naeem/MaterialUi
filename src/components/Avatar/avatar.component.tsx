import * as React from 'react';
import { BlueRainType, withBlueRain } from '@blueeast/bluerain-os';

/**
 * The universal props of Avatar Component
 * @param {Object} props.style Get the style object.
 * @param {string} props.src src for Avatar.
 * @param {Node} props.children children of Avatar.
 */

export interface IAvatar {
	children?: React.ReactNode;
	size?: number,
	src?: string,
	style?: {},
}
/**
 * The universal props of Avatar Component
 * @param {Object} props.style Get the style object.
 * @param {string} props.src src for Avatar.
 * @param {Node} props.children children of Avatar.
 *  The specific libraray props of Badge Component
 * @param {string} props.alt The enum color of Badge.
 * @param {Object} props.classes the children on which the badge will append styles.
 * @param {string or function} props.component The enum color of Badge.
 * @param {Object} props.imgProps the children on which the badge will append styles.
 * @param {string} props.sizes The sizes of image.
 */

export interface IAvatarProps {
	alt?: string,
	children?: React.ReactNode;
	classes?: {},
	component?: any,
	imgProps?: {},
	sizes?: string,
	src?: string,
	style?: {},
	bluerain?: BlueRainType,
}
const BlueRainAvatar = (props: IAvatar & IAvatarProps) => {
	const size = props.size? props.size: 70;
	const avatarStyle = [{ width:size, height:size, backgroundColor:'lightgrey', borderRadius:1000, display:'flex', justifyContent:'center', alignItems:'center', fontSize:.5*size }, { ...props.style }];
	const View = props.bluerain ? props.bluerain.Components.get('View') : 'div';
	const Text = props.bluerain ? props.bluerain.Components.get('Text') : 'div';
	const Image = props.bluerain ? props.bluerain.Components.get('Image') : 'img';
	if (props.children) {
		return <View style={avatarStyle}>{props.children}</View>;
	}
	return <Image source={props.src} style={avatarStyle} />;

};

export default withBlueRain(BlueRainAvatar);
