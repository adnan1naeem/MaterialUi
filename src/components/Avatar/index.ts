// export { default as Avatar, MUIAvatarProperties } from './Avatar.component';
