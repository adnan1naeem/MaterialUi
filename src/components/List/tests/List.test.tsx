import { List, ListSubheader, ListItem } from '../';
import { Avatar } from '../../Avatar';
import renderer from 'react-test-renderer';
import React from 'react';


test('List component with style props', () => {
	const list = renderer.create(
		<List style={{ width: 100 }}></List>
	).toJSON();
	expect(list.props.style).toEqual({ width: 100 });
	expect(list).toMatchSnapshot();
});

test('List component with children props', () => {
	const list = renderer.create(
		<List><div>Hello</div></List>
	).toJSON();
	expect(list.props.children).toEqual(undefined);
	expect(list).toMatchSnapshot();
});

test('ListSuheader with children props', () => {
	const listheader = renderer.create(
		<ListSubheader>I am Header of list</ListSubheader>
	).toJSON();
	expect(listheader.props.children).toEqual(undefined);
	expect(listheader).toMatchSnapshot();
});

test('ListItem  with children props', () => {
	const listitem = renderer.create(
		<ListItem>Hello</ListItem>
	).toJSON();
	expect(listitem.props.children).toEqual(undefined);
	expect(listitem).toMatchSnapshot();
});


test('ListItem with   style props', () => {
	const listitem = renderer.create(
		<ListItem style={{ width: 100 }} />
	).toJSON();
	expect(listitem.props.style).toEqual({ width: 100 });
	expect(listitem).toMatchSnapshot();
});

test('ListItem with primary props', () => {
	const listitem = renderer.create(
		<ListItem primary="hello">Hello</ListItem>
	).toJSON();
	expect(listitem.props.primary).toEqual('hello');
	expect(listitem).toMatchSnapshot();
});

test('ListItem with secondary props', () => {
	const listitem = renderer.create(
		<ListItem secondary="Hi">Hello</ListItem>
	).toJSON();
	expect(listitem.props.secondary).toEqual('Hi');
	expect(listitem).toMatchSnapshot();
});
