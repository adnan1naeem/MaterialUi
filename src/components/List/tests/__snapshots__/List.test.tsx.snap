// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component with children props 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
  style={undefined}
>
  <div>
    Hello
  </div>
</ul>
`;

exports[`List component with style props 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
  style={
    Object {
      "width": 100,
    }
  }
/>
`;

exports[`ListItem  with children props 1`] = `
<li
  className="MuiListItem-root-10 MuiListItem-gutters-17 MuiListItem-default-13"
  disabled={false}
  onClick={undefined}
>
  <div
    className="MuiListItemText-root-20"
  >
    <h3
      className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-text-23"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-49"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
    </h3>
  </div>
</li>
`;

exports[`ListItem with   style props 1`] = `
<li
  className="MuiListItem-root-10 MuiListItem-gutters-17 MuiListItem-default-13"
  disabled={false}
  onClick={undefined}
  style={
    Object {
      "width": 100,
    }
  }
/>
`;

exports[`ListItem with action props 1`] = `
<div
  className="MuiListItem-container-11"
>
  <li
    action={<getContext(BlueRainAvatar) />}
    className="MuiListItem-root-10 MuiListItem-gutters-17 MuiListItem-secondaryAction-19 MuiListItem-default-13"
    disabled={false}
    onClick={undefined}
  >
    <div
      className="MuiListItemText-root-20"
    >
      <h3
        className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-text-23"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-49"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
          />
        </svg>
      </h3>
    </div>
  </li>
  <div
    className="MuiListItemSecondaryAction-root-56"
  >
    <img
      source={undefined}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "lightgrey",
            "borderRadius": 1000,
            "display": "flex",
            "fontSize": 35,
            "height": 70,
            "justifyContent": "center",
            "width": 70,
          },
          Object {},
        ]
      }
    />
  </div>
</div>
`;

exports[`ListItem with avatar props 1`] = `
<li
  avatar={<getContext(BlueRainAvatar) />}
  className="MuiListItem-root-10 MuiListItem-gutters-17 MuiListItem-default-13"
  disabled={false}
  onClick={undefined}
>
  <img
    source={undefined}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 1000,
          "display": "flex",
          "fontSize": 35,
          "height": 70,
          "justifyContent": "center",
          "width": 70,
        },
        Object {},
      ]
    }
  />
</li>
`;

exports[`ListItem with primary props 1`] = `
<li
  className="MuiListItem-root-10 MuiListItem-gutters-17 MuiListItem-default-13"
  disabled={false}
  onClick={undefined}
  primary="hello"
>
  <div
    className="MuiListItemText-root-20"
  >
    <h3
      className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-text-23"
    >
      hello
    </h3>
    
  </div>
  <div
    className="MuiListItemText-root-20"
  >
    <h3
      className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-text-23"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-49"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
    </h3>
  </div>
</li>
`;

exports[`ListItem with secondary props 1`] = `
<li
  className="MuiListItem-root-10 MuiListItem-gutters-17 MuiListItem-default-13"
  disabled={false}
  onClick={undefined}
  secondary="Hi"
>
  <div
    className="MuiListItemText-root-20"
  >
    
    <p
      className="MuiTypography-root-25 MuiTypography-body1-34 MuiTypography-colorSecondary-46 MuiListItemText-text-23"
    >
      Hi
    </p>
  </div>
  <div
    className="MuiListItemText-root-20"
  >
    <h3
      className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-text-23"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-49"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
    </h3>
  </div>
</li>
`;

exports[`ListSuheader with children props 1`] = `
<li
  className="MuiListSubheader-root-5 MuiListSubheader-sticky-9"
>
  I am Header of list
</li>
`;
